extras/Catch.cmake

```diff
@@ -249,7 +259,15 @@ function(catch_discover_tests TARGET)
       "endif()"                                                                    "\n"
     )
 
-    file(GENERATE OUTPUT "${ctest_include_file}" CONTENT "${ctest_include_content}")
+    if(GENERATOR_IS_MULTI_CONFIG)
+      foreach(_config ${CMAKE_CONFIGURATION_TYPES})
+        file(GENERATE OUTPUT "${ctest_file_base}_include-${_config}.cmake" CONTENT "${ctest_include_content}" CONDITION $<CONFIG:${_config}>)
+      endforeach()
+      file(WRITE "${ctest_include_file}" "include(\"${ctest_file_base}_include-\${CTEST_CONFIGURATION_TYPE}.cmake\")")
```

Comment:
I would love to see this dummy file check that `CTEST_CONFIGURATION_TYPE` is set, and if it is not, providing a nice error message instead of this beauty

```
ctest
Test project C:/ubuntu/temp/discover-tests-tests/msvc-sln-1
CMake Error at C:/ubuntu/temp/discover-tests-tests/msvc-sln-1/tests-b12d07c_include.cmake:1 (include):
  include could not find requested file:

    C:/ubuntu/temp/discover-tests-tests/msvc-sln-1/tests-b12d07c_include-.cmake
Call Stack (most recent call first):
  CTestTestfile.cmake:7 (include)


Errors while running CTest
```

(This is not needed for merging, but it would be nice 😃 )

Reply:
Great, thanks.